modules/react-map-gl-draw/src/mode-handler.js

```diff
@@ -0,0 +1,416 @@
+// @flow
+import { _MapContext as MapContext } from 'react-map-gl';
+import React, { Component } from 'react';
+import { ImmutableFeatureCollection } from '@nebula.gl/edit-modes';
+
+import type { Position, EditAction } from '@nebula.gl/edit-modes';
+import type { MjolnirEvent } from 'mjolnir.js';
+import type { BaseEvent, EditorProps, EditorState, Mode } from './types';
+
+import { DRAWING_MODE, EDIT_TYPE, MODES } from './constants';
+import { getScreenCoords, isNumeric, parseEventElement } from './edit-modes/utils';
+import {
+  BaseMode,
+  EditingMode,
+  DrawPointMode,
+  DrawLineStringMode,
+  DrawRectangleMode,
+  DrawPolygonMode
+} from './edit-modes';
```

Comment:
Shouldn't this be `@nebula.gl/edit-modes` ?

Reply:
This is step 1 of refactoring, I need to make sure the rendering of react-map-gl-draw can handle the APIs and data flow of `@nebula.gl/edit-mode`. After Clay finished his refactoring and landed his PRs, I'll make another pass to extend from there.